src/compiler/declarationEmitter.ts

```diff
@@ -358,7 +358,7 @@ namespace ts {
             }
             else {
                 errorNameNode = declaration.name;
-                const format = TypeFormatFlags.UseTypeOfFunction | TypeFormatFlags.WriteDefaultSymbolWithoutName |
+                const format = TypeFormatFlags.UseTypeOfFunction | TypeFormatFlags.UseStructuralFallback | TypeFormatFlags.WriteDefaultSymbolWithoutName |

```

Comment:
It looks like we're using `TypeFormatFlags.UseStructuralFallback` in all the same places we already use `TypeFormatFlags.UseTypeOfFunction`. Can they be combined?

Reply:
They could be, but then external consumers lose the choice of allowing the structural fallback or not. I'm erring on the side of being more configurable here, since they are separate behaviors. This way allows something like a quickfix to try to build a node using typeof with no fallback, see if there's an error, than decide not to offer the fix, for example.